iclaims.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2021-2022 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package psatoken
+
+import (
+	"errors"
+	"fmt"
+)
+
+// IClaims provides a uniform interface for dealing with claims in all supported
+// profiles
+type IClaims interface {
+	// Getters
+	GetProfile() (string, error)
+	GetClientID() (int32, error)
+	GetSecurityLifeCycle() (uint16, error)
+	GetImplID() ([]byte, error)
+	GetBootSeed() ([]byte, error)
+	GetCertificationReference() (string, error)
+	GetSoftwareComponents() ([]SwComponent, error)
+	GetNonce() ([]byte, error)
+	GetInstID() ([]byte, error)
+	GetVSI() (string, error)
+
+	// Setters
+	SetClientID(int32) error
+	SetSecurityLifeCycle(uint16) error
+	SetImplID([]byte) error
+	SetBootSeed([]byte) error
+	SetCertificationReference(string) error
+	SetSoftwareComponents([]SwComponent) error
+	SetNonce([]byte) error
+	SetInstID([]byte) error
+	SetVSI(string) error
+
+	// CBOR codecs
+	FromCBOR([]byte) error
+	ToCBOR() ([]byte, error)
+
+	// Semantic validation
+	Validate() error
+}
+
+func NewClaims(profile string) (IClaims, error) {
+	switch profile {
+	case PsaProfile1:
+		includeProfileClaim := true
```

Comment:
I am not clear is there a need for includeProfileClaim? Can it not mirror the newP2Claims() ?

Reply:
Including the profile claim in the claims-set is optional in P1 and mandatory in P2.
That's the reason for the difference in signature.

Reply:
Sure thanks! Then this flexibility should be provided to the external caller for the API? If it wants to include the Profile claim when encoding ?

Reply:
I don't think this flexibility is necessary.  In fact, I think allowing the profile to be optional is an error in the old spec.

Reply:
I disagree. If we have a spec that is now public, we should comply with the specification. I am little confused here.

Can we please discuss this and then resolve this on Monday ?

Reply:
> I disagree. If we have a spec that is now public, we should comply with the specification.

we do; there is nothing here that goes against the spec.  We only take the liberty to *always* produce an optional claim, that's it.

Reply:
what the spec says is that *the receiver* must be able to understand a claims-set with the profile claim missing.  It doesn't say anything about the behaviour of *the producer* wrt this.

Reply:
resolved f2f